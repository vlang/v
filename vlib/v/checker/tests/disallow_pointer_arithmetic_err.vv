fn main() {
	x := 5
	p := &x
	_ := p + p // should be error
	_ := p * p // should be error
	_ := p * 2 // should be error
	_ := p + 5 // OK but only in unsafe block, r is *int
	_ := p - p // OK even in safe code, but n should be isize
}
