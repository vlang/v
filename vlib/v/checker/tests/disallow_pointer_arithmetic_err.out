vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:7:7: warning: pointer arithmetic is only allowed in `unsafe` blocks
    5 |     _ := p * p // should be error
    6 |     _ := p * 2 // should be error
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
      |          ~~~~~
    8 |     _ := p - p // OK even in safe code, but n should be isize
    9 | }
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:8:7: warning: pointer arithmetic is only allowed in `unsafe` blocks
    6 |     _ := p * 2 // should be error
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
    8 |     _ := p - p // OK even in safe code, but n should be isize
      |          ~~~~~
    9 | }
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:4:7: error: invalid operator `+` to `&int` and `&int`
    2 |     x := 5
    3 |     p := &x
    4 |     _ := p + p // should be error
      |          ~~~~~
    5 |     _ := p * p // should be error
    6 |     _ := p * 2 // should be error
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:5:7: error: invalid operator `*` to `&int` and `&int`
    3 |     p := &x
    4 |     _ := p + p // should be error
    5 |     _ := p * p // should be error
      |          ~~~~~
    6 |     _ := p * 2 // should be error
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:5:7: error: infix `*` is not defined for pointer values
    3 |     p := &x
    4 |     _ := p + p // should be error
    5 |     _ := p * p // should be error
      |          ~~~~~
    6 |     _ := p * 2 // should be error
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:6:7: error: invalid operator `*` to `&int` and `int literal`
    4 |     _ := p + p // should be error
    5 |     _ := p * p // should be error
    6 |     _ := p * 2 // should be error
      |          ~~~~~
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
    8 |     _ := p - p // OK even in safe code, but n should be isize
vlib/v/checker/tests/disallow_pointer_arithmetic_err.vv:6:7: error: infix `*` is not defined for pointer values
    4 |     _ := p + p // should be error
    5 |     _ := p * p // should be error
    6 |     _ := p * 2 // should be error
      |          ~~~~~
    7 |     _ := p + 5 // OK but only in unsafe block, r is *int
    8 |     _ := p - p // OK even in safe code, but n should be isize
